### ISSUE TITLE
[Bug] aria2c error on Unicode character

### ISSUE THREAD

**Issue by pukkandan** [pukkandan/yt-dlp#71]
```
Exception: [../../aria2-release-1.35.0/src/AbstractOptionHandler.cc:69] errorCode=28 We encountered a problem while processing the option '--console-log-level'.
  -> [../../aria2-release-1.35.0/src/OptionHandlerImpl.cc:435] errorCode=1 console-log-level must be one of the following:'debug' 'info' 'notice' 'warn' 'error'
Usage:
 --console-log-level=LEVEL    Set log level to output to console.
                              Possible Values: debug, info, notice, warn, error
                              Default: notice
                              Tags: #advanced
[download] Finished downloading playlist: 1
```

_Originally posted by @candrapersada in https://github.com/pukkandan/yt-dlp/issues/55#issuecomment-776899332_

**Comment by pukkandan** [comment]
@candrapersada Please give the full verbose log

**Comment by candrapersada** [comment]
```
youtube-dlc --newline -i --hls-prefer-native --external-downloader aria2c --external-downloader-args "--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16" -o E:/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s --cookies D:/Tartube-data/cookies.txt -f bestvideo+bestaudio --download-archive E:/ytdl-archive.txt https://www.youtube.com/playlist?list=PLN5V6aiYlgnvVxc9_QcD_3BVSbeRw5vr4
[youtube:tab] PLN5V6aiYlgnvVxc9_QcD_3BVSbeRw5vr4: Downloading webpage
[download] Downloading playlist: 1
[youtube:tab] playlist 1: Downloading 26 videos
[download] Downloading video 1 of 26
[youtube] s3ErYKn70Zw: Downloading webpage
[youtube] s3ErYKn70Zw: Downloading MPD manifest
[info] Downloading format(s) 244+140
[dashsegments] Total fragments: 193
[download] Destination: E:/1/01 - Phase 0 .f244.webm
[download] Downloading video 2 of 26
[youtube] 27OTb5FPqdE: Downloading webpage
[info] Downloading format(s) 244+251
[download] Destination: E:/1/02 - Phase 1-1 .f244.webm
Exception: [../../aria2-release-1.35.0/src/AbstractOptionHandler.cc:69] errorCode=28 We encountered a problem while processing the option '--console-log-level'.
  -> [../../aria2-release-1.35.0/src/OptionHandlerImpl.cc:435] errorCode=1 console-log-level must be one of the following:'debug' '
```

**Comment by pukkandan** [comment]
```$ youtube-dlc --external-downloader aria2c https://www.youtube.com/playlist?list=PLN5V6aiYlgnvVxc9_QcD_3BVSbeRw5vr4 -v --playlist-items 1
[debug] Command-line config: ['--external-downloader', 'aria2c', 'https://www.youtube.com/playlist?list=PLN5V6aiYlgnvVxc9_QcD_3BVSbeRw5vr4', '-v', '--playlist-items', '1']
[debug] Loading archive file None
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.02.09
[debug] Plugin Extractors: ['SamplePlugin']
[debug] Git HEAD: 8d801631c
[debug] Python version 3.8.4 (CPython) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-11-19-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-19-full_build-www.gyan.dev
[debug] Proxy map: {}
[youtube:tab] PLN5V6aiYlgnvVxc9_QcD_3BVSbeRw5vr4: Downloading webpage
[download] Downloading playlist: アルトネリコ1
[youtube:tab] playlist アルトネリコ1: Downloading 1 videos
[download] Downloading video 1 of 1
[youtube] s3ErYKn70Zw: Downloading webpage
[youtube] s3ErYKn70Zw: Downloading MPD manifest
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9(9), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] Downloading format(s) 244+140
WARNING: Requested formats are incompatible for merge and will be merged into mkv.
[debug] Invoking downloader on 'https://manifest.googlevideo.com/api/manifest/dash/expire/1613005337/ei/uS0kYPvVJOGq8QOnlpG4CA/ip/2405%3A201%3Af003%3Ab167%3A357a%3Ab92d%3A59b8%3A8b3a/id/b3712b60a9fbd19c/source/youtube/requiressl/yes/playback_host/r1---sn-gwpa-jv36.googlevideo.com/mh/cG/mm/31%2C29/mn/sn-gwpa-jv36%2Csn-gwpa-h55r/ms/au%2Crdu/mv/m/mvi/1/pl/38/hfr/all/as/fmp4_audio_clear%2Cwebm_audio_clear%2Cwebm2_audio_clear%2Cfmp4_sd_hd_clear%2Cwebm2_sd_hd_clear/initcwndbps/253750/vprv/1/mt/1612983399/fvip/1/keepalive/yes/itag/0/sparams/expire%2Cei%2Cip%2Cid%2Csource%2Crequiressl%2Chfr%2Cas%2Cvprv%2Citag/sig/AOq0QJ8wRAIgEX6UtsO6okKGRWPyUtdIP7Fk_tvRBHxwVs_86ZJHUbMCIFVkTd4BVcjAUwN8Q6xQhDfHIkKe0EAARlP7a-h4-BAS/lsparams/playback_host%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps/lsig/AG3C_xAwRAIgWAuzq57p68-nhhUmjoKDua2UrfbphfnmuDk9DQw0uqoCIBs5hc4iVl6khdaK4aTc67-Wk16NbZG5cpxGMTfftSD2'
[dashsegments] Total fragments: 193
[download] Destination: アルトネリコ　Phase 0 序 [s3ErYKn70Zw].f244.webm
ERROR: 'charmap' codec can't encode characters in position 751-757: character maps to <undefined>
Traceback (most recent call last):
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 1004, in wrapper
    return func(self, *args, **kwargs)
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 1312, in __process_iterable_entry
    return self.process_ie_result(
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 1110, in process_ie_result
    return self.process_ie_result(
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 1073, in process_ie_result
    return self.process_video_result(ie_result, download=download)
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 1943, in process_video_result
    self.process_info(new_info)
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 2347, in process_info
    partial_success, real_download = dl(fname, new_info)
  File "<REDACTED>\youtube_dlc\YoutubeDL.py", line 2120, in dl
    return fd.download(name, info, subtitle)
  File "<REDACTED>\youtube_dlc\downloader\common.py", line 375, in download
    return self.real_download(filename, info_dict), True
  File "<REDACTED>\youtube_dlc\downloader\dash.py", line 98, in real_download
    success = fd.real_download(filename, info_copy)
  File "<REDACTED>\youtube_dlc\downloader\external.py", line 47, in real_download
    retval = self._call_downloader(tmpfilename, info_dict)
  File "<REDACTED>\youtube_dlc\downloader\external.py", line 114, in _call_downloader
    cmd = [encodeArgument(a) for a in self._make_cmd(tmpfilename, info_dict)]
  File "<REDACTED>\youtube_dlc\downloader\external.py", line 267, in _make_cmd
    f.write('\n'.join(url_list))
  File "<REDACTED>\Python\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 751-757: character maps to <undefined>
```

@shirt, can you have a look

Possibly related: https://www.reddit.com/r/youtubedl/comments/lgconf/ytdlp_release_20210209/gmtp1aa?context=3

**Comment by pukkandan** [comment]
@candrapersada Since the error message you got is different from mine, could you confirm that your issue is fixed in current master?

**Comment by candrapersada** [comment]
```
[debug] Loading archive file '/home/candra/Unduhan/Folder Tanpa Judul/ytdl-archive.txt'
[youtube] bDheuXvSNns: Downloading webpage
[youtube] bDheuXvSNns: Downloading MPD manifest
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9(9), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Downloading format(s) 160+139
[debug] Invoking downloader on 'https://manifest.googlevideo.com/api/manifest/dash/expire/1613031301/ei/JZMkYMG2ELHU3LUP-byXiAM/ip/103.105.78.11/id/6c385eb97bd2367b/source/youtube/requiressl/yes/playback_host/r2---sn-poqvn5u-jb3z.googlevideo.com/mh/Wa/mm/31%2C26/mn/sn-poqvn5u-jb3z%2Csn-30a7rn76/ms/au%2Conr/mv/m/mvi/2/pl/24/hfr/all/as/fmp4_audio_clear%2Cwebm_audio_clear%2Cwebm2_audio_clear%2Cfmp4_sd_hd_clear%2Cwebm2_sd_hd_clear/initcwndbps/81250/vprv/1/mt/1613009329/fvip/2/keepalive/yes/itag/0/sparams/expire%2Cei%2Cip%2Cid%2Csource%2Crequiressl%2Chfr%2Cas%2Cvprv%2Citag/sig/AOq0QJ8wRgIhALpZls5k8I6VhSzJzi7ONnPEXQF8j6H_4E4bxwtV0XsQAiEArddpifF56wsSfPZOGb8mjgTCEU6UNiq2SmslpE3Oio0%3D/lsparams/playback_host%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps/lsig/AG3C_xAwRAIgB5n04Wim4CarStFLNcGoUqHUFSix3uLxrXmql8-UB4kCIF-juDGHL-64nKAehXstSfWXbB-JBWuY8WrOz26bFNgY'
[dashsegments] Total fragments: 196
[download] Destination: /home/candra/Unduhan/Folder Tanpa Judul/NA/NA - テイルズ オブ グレイセス エフ ストーリーパート part2.f160.mp4
[debug] aria2c command line: aria2c -c '--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16' --dir '/home/candra/Unduhan/Folder Tanpa Judul/NA' --header 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3519.2 Safari/537.36' --header 'Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'Accept-Encoding: gzip, deflate' --header 'Accept-Language: en-us,en;q=0.5' --check-certificate=true --remote-time=true --auto-file-renaming=false --console-log-level=warn --summary-interval=0 --uri-selector inorder --download-result=hide -i '/home/candra/Unduhan/Folder Tanpa Judul/NA/NA - テイルズ オブ グレイセス エフ ストーリーパート part2.f160.mp4.part.frag.urls'
Exception: [AbstractOptionHandler.cc:69] errorCode=28 We encountered a problem while processing the option '--console-log-level'.
  -> [OptionHandlerImpl.cc:435] errorCode=1 console-log-level must be one of the following:'debug' 'info' 'notice' 'warn' 'error'
Usage:
 --console-log-level=LEVEL    Set log level to output to console.
                              Nilai-nilai yang dimungkinkan: debug, info, notice, warn, error
                              Default: notice
                              Tag: #advanced
[debug] Command-line config: ['--newline', '-i', '--hls-prefer-native', '--external-downloader', 'aria2c', '--external-downloader-args', '"--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16"', '-o', '/home/candra/Unduhan/Folder Tanpa Judul/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s', '--cookies', '/media/candra/Game/Tartube-data-Linux/cookies.txt', '-f', 'worstvideo+worstaudio/worst', '-v', '--download-archive', '/home/candra/Unduhan/Folder Tanpa Judul/ytdl-archive.txt', '--ffmpeg-location', '/media/candra/Game/FFmpeg/ffmpeg', 'https://youtu.be/bDheuXvSNns']
WARNING: --external-downloader-args given without specifying name. The arguments will be given to all external downloaders
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.02.09
[debug] Python version 3.8.5 (CPython) - Linux-5.4.0-64-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.3.1-static, ffprobe 4.3.1-static
[debug] Proxy map: ()

```

**Comment by pukkandan** [comment]
I don't see a `git head` in your verbose log. You are probably still using the last release and not the master branch.

Btw, why is the log scrambled?

**Comment by pukkandan** [comment]
> [debug] Command-line config: ['--newline', '-i', '--hls-prefer-native', '--external-downloader', 'aria2c', '--external-downloader-args', '"--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16"', '-o', '/home/candra/Unduhan/Folder Tanpa Judul/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s', '--cookies', '/media/candra/Game/Tartube-data-Linux/cookies.txt', '-f', 'worstvideo+worstaudio/worst', '-v', '--download-archive', '/home/candra/Unduhan/Folder Tanpa Judul/ytdl-archive.txt', '--ffmpeg-location', '/media/candra/Game/FFmpeg/ffmpeg', 'https://youtu.be/bDheuXvSNns']

Found the issue - There seem to be an extra `""` around `--downloader-args`. What is the command you ran?

**Comment by candrapersada** [comment]
yes

**Comment by pukkandan** [comment]
> What is the command you ran?

**Comment by candrapersada** [comment]
`python3 /media/candra/Game/youtube-dl(c)/youtube-dlc --newline -i --hls-prefer-native --external-downloader aria2c --external-downloader-args "--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16" -o /home/candra/Video/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s --cookies /media/candra/Game/Tartube-data-Linux/cookies.txt -f worstvideo+worstaudio/worst -v --download-archive /home/candra/Video/ytdl-archive.txt --ffmpeg-location /media/candra/Game/FFmpeg/ffmpeg https://youtu.be/bDheuXvSNns`

**Comment by pukkandan** [comment]
That is most definitely not the command you are running. You'll just get a bash syntax error :/

```
$ python3 -m youtube_dlc --newline -i --hls-prefer-native --external-downloader aria2c --external-downloader-args "--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16" -o Video/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s --cookies cookies.txt -f worstvideo+worstaudio/worst -v --download-archive Video/ytdl-archive.txt --ffmpeg-location /media/candra/Game/FFmpeg/ffmpeg https://youtu.be/bDheuXvSNns
bash: syntax error near unexpected token `('
```

**Comment by candrapersada** [comment]
```
python3 /home/candra/Unduhan/youtube-dlc --newline -i --hls-prefer-native --external-downloader aria2c --external-downloader-args "--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16" -o /home/candra/Video/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s --cookies /media/candra/Game/Tartube-data-Linux/cookies.txt -f worstvideo+worstaudio/worst -v --download-archive /home/candra/Video/ytdl-archive.txt --ffmpeg-location /media/candra/Game/FFmpeg/ffmpeg https://youtu.be/bDheuXvSNns
[debug] Loading archive file '/home/candra/Video/ytdl-archive.txt'
[youtube] bDheuXvSNns: Downloading webpage
[youtube] bDheuXvSNns: Downloading MPD manifest
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9(9), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Downloading format(s) 160+139
[debug] Invoking downloader on 'https://manifest.googlevideo.com/api/manifest/dash/expire/1613091739/ei/O38lYJSpOOjFz7sPjYGpmAc/ip/103.105.78.11/id/6c385eb97bd2367b/source/youtube/requiressl/yes/playback_host/r2---sn-poqvn5u-jb3z.googlevideo.com/mh/Wa/mm/31%2C26/mn/sn-poqvn5u-jb3z%2Csn-i3belnl7/ms/au%2Conr/mv/m/mvi/2/pl/24/hfr/all/as/fmp4_audio_clear%2Cwebm_audio_clear%2Cwebm2_audio_clear%2Cfmp4_sd_hd_clear%2Cwebm2_sd_hd_clear/initcwndbps/98750/vprv/1/mt/1613069819/fvip/2/keepalive/yes/itag/0/sparams/expire%2Cei%2Cip%2Cid%2Csource%2Crequiressl%2Chfr%2Cas%2Cvprv%2Citag/sig/AOq0QJ8wRgIhAP1AyFwNz0pFjqNew1f0nAfxg_zm3wzTWWBouVia0iaEAiEA-h0opGMcXbOSq0p-17eo1tZs43xVETOxBgHf-tVNKBA%3D/lsparams/playback_host%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps/lsig/AG3C_xAwRQIhAJvhEvYuG7G4z6CSUWVzC6ZcYhkqDfYvp2oviKQRp5D1AiAVjVnpkXLN42hcp3rdp4jwpHC61l-_ePQPO34guSC46w%3D%3D'
[dashsegments] Total fragments: 196
[download] Destination: /home/candra/Video/NA/NA - テイルズ オブ グレイセス エフ ストーリーパート part2.f160.mp4
[debug] aria2c command line: aria2c -c '--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16' --dir /home/candra/Video/NA --header 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3779.1 Safari/537.36' --header 'Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7' --header 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8' --header 'Accept-Encoding: gzip, deflate' --header 'Accept-Language: en-us,en;q=0.5' --check-certificate=true --remote-time=true --auto-file-renaming=false --console-log-level=warn --summary-interval=0 --uri-selector inorder --download-result=hide -i '/home/candra/Video/NA/NA - テイルズ オブ グレイセス エフ ストーリーパート part2.f160.mp4.part.frag.urls'
Exception: [AbstractOptionHandler.cc:69] errorCode=28 We encountered a problem while processing the option '--console-log-level'.
  -> [OptionHandlerImpl.cc:435] errorCode=1 console-log-level must be one of the following:'debug' 'info' 'notice' 'warn' 'error'
Usage:
 --console-log-level=LEVEL    Set log level to output to console.
                              Nilai-nilai yang dimungkinkan: debug, info, notice, warn, error
                              Default: notice
                              Tag: #advanced
[debug] Command-line config: ['--newline', '-i', '--hls-prefer-native', '--external-downloader', 'aria2c', '--external-downloader-args', '"--console-log-level=warn --summary-interval=0 --file-allocation=none -x16 -j16 -s16"', '-o', '/home/candra/Video/%(playlist)s/%(playlist_index)s - %(title)s.%(ext)s', '--cookies', '/media/candra/Game/Tartube-data-Linux/cookies.txt', '-f', 'worstvideo+worstaudio/worst', '-v', '--download-archive', '/home/candra/Video/ytdl-archive.txt', '--ffmpeg-location', '/media/candra/Game/FFmpeg/ffmpeg', 'https://youtu.be/bDheuXvSNns']
WARNING: --external-downloader-args given without specifying name. The arguments will be given to all external downloaders
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.02.09
[debug] Python version 3.8.5 (CPython) - Linux-5.4.0-64-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.3.1-static, ffprobe 4.3.1-static
[debug] Proxy map: ()

```

**Comment by pukkandan** [comment]
@candrapersada Does the issue still exist in the latest release?

**Comment by candrapersada** [comment]
how to use the correct arguments for aria2 I ask for an example?

**Comment by pukkandan** [comment]
If you want to use the default arguments, u can just use `--external-downloader aria2c`

If you want to use custom args, I reccomend using `--external-downloader aria2c --downloader-args "aria2c:<NEW ARGS>"`  

But `--external-downloader aria2c --external-downloader-args "<NEW ARGS>"` also work for backward compatibility

**Comment by pukkandan** [comment]
@candrapersada please confirm whether or not the issue has been resolved for you so that I can close the issue

**Comment by candrapersada** [comment]
don't close it first because you have to wait for issues from #34 to finish too.

**Comment by pukkandan** [comment]
what does it have to do with this issue?

**Comment by candrapersada** [comment]
for GUI from Tartube

**Comment by shirt-dev** [comment]
> for yt-dlp GUI from Tartube

That is not related to this issue, if you experience issues after #34 is closed it can be reopened.
